tableaudocumentapi/datasource.py

```diff
@@ -145,21 +150,23 @@ def __init__(self, dsxml, filename=None):
 
     @classmethod
     def from_file(cls, filename):
-        """Initialize datasource from file (.tds)"""
+        """Initialize datasource from file (.tds ot .tdsx)"""
 
-        dsxml = xml_open(filename, cls.__name__.lower()).getroot()
+        dsxml = xml_open(filename, 'datasource').getroot()
```

Comment:
I realized that this was actually less readable than just making it a literal string, which removed the need for the comment "#Gets the name of the class and normalizes it so it can verify the file it opens"